### DISCUSSION TITLE
Oriented bounding box

### DISCUSSION THREAD

**Discussion by johnnybigert** [elalish/manifold#1661] [Q&A]
Hi! Are there any functions to get the oriented (minimal) bounding box? I looked through the Python bindings but found only the normal (axis-aligned) bounding box.

If not, could you please consider adding it? Thanks.

**Comment by elalish** [comment]
Nope - care to add it?

**Comment by Magnolia2423** [comment]
I’m thinking of adding oriented bounding box support.
Would a PCA-based OBB be acceptable as a first step,
or would you prefer an exact minimum-volume solution?

**Comment by hrgdavor** [comment]
That is something I was exploring, but I suck at computational geometry so bad ... 

@Magnolia2423  care to share some resources, or point to directions about oriented bounding boxes. Ofc I can google it, but it is always nice to get pointers from someone that knows more on a topic.

**Comment by Magnolia2423** [comment]
Totally fair — OBBs get gnarly fast 🙂

A PCA-based OBB is usually the easiest place to start:
- Compute the covariance of vertex positions
- Use the eigenvectors as the box axes
- Project vertices onto these axes to determine the extents

It’s not guaranteed to be minimum-volume, but it’s fast and often good enough.

For exact or closer-to-minimum solutions:
- The usual approach is to compute the convex hull
- Then evaluate candidate orientations derived from face normals / edges
- Conceptually similar to rotating calipers, but in 3D (so more complex)

Some good pointers:
- “Minimum Volume Bounding Box” by O’Rourke (classic reference)
- Open3D’s `get_minimal_oriented_bounding_box`
- CGAL’s `oriented_bounding_box` (exact, but heavier)

My idea was to start with a PCA-based OBB and iterate from there if needed.
Happy to help with implementation details.

**Comment by pca006132** [comment]
> Ofc I can google it, but it is always nice to get pointers from someone that knows more on a topic.

From the previous PR, it seems to me that @Magnolia2423 just uses AI without checking anything.

Anyway, in case someone wants to do this, we do not accept adding dependencies. AI generated code is fine, but at least read your code before sending a PR.

**Comment by zmerlynn** [comment]
Do y'all have thoughts/opinions on an API here? Seems like it could be as small as a `mat3x4` that you would apply to a unit cube, all the way to defining a new `OrientedBox` type with a similar-ish API to `Box`. (Leaning hard away from modifying `Box` itself due to memory cost.) Could start with the least convenient / smallest surface (transform) and grow from there.

**Comment by elalish** [comment]
I don't see any reason to add OBBs to manifold. We don't need them and there are plenty of math libraries people can use for them if they need.

**Comment by gstreetmedia** [comment]
If you were working with Three.js, I created an OrientedBox3, which is pretty useful when combined with Manifold. I use it to find out if things actually intersect (generally), before running a manifold Boolean operation. It's boat loads faster than, say doing an intersection operation, to find out if there actually is an intersection. 

https://gist.github.com/gstreetmedia/f9dae7e50fbc3d3eb5b6dbb24e961d1c
